src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/RazorSemanticTokensInfoService.cs

```diff
@@ -142,35 +142,28 @@ private static ImmutableArray<SemanticRange> CombineSemanticRanges(ImmutableArra
         string? previousResultId = null)
     {
         var generatedDocument = codeDocument.GetCSharpDocument();
-        Range[]? csharpRanges;
 
-        // When the feature flag is enabled we try to get a list of precise ranges for the C# code embedded in the Razor document.
-        // The feature flag allows to make calls to Roslyn using multiple smaller and disjoint ranges of the document
-        if (_languageServerFeatureOptions.UsePreciseSemanticTokenRanges)
+        // When the feature flag is disabled, we fallback to computing a single range for the entire document.
+        // This single range is the minimal range that contains all of the C# code in the document.
+        // We'll try to call into the mapping service to map to the projected range for us. If that doesn't work,
+        // we'll try to find the minimal range ourselves.
+        if (!_documentMappingService.TryMapToGeneratedDocumentRange(generatedDocument, razorRange, out var minimalRange) &&
```

Comment:
I think this might mean the removal of the fix the previous code made, where we used to not classify the string portion of addTagHelper correctly. Would be an extra bonus if the new system was able to provide better classification, not just faster

Reply:
- [x] Brought back the fix the previous code made, where we used to not classify the string portion of addTagHelper correctly